controller/scheduler-v2/scheduler.go

```diff
@@ -135,88 +171,102 @@ func (s *Scheduler) Sync() (err error) {
 				}
 				if _, ok := s.jobs[job.ID]; ok {
 					log.Info("skipping known job")
+					delete(inactiveJobs, job.ID)
 					continue
 				}
 
 				log.Info("getting formation")
-				f, err := s.getFormation(appID, appName, releaseID)
-				if err != nil {
-					log.Error("error getting formation", "err", err)
+				f := s.formations.Get(appID, releaseID)
+				if f == nil {
+					err = fmt.Errorf("No such formation obtained")
```

Comment:
This should probably just be a flag which triggers the formation sync:

```go
unknownFormations := false
defer func() {
	if unknownFormations {
		s.formationSync <- struct{}{}
	}
}()
```

and thinking about this, we probably want to trigger another job sync straight after the triggered formation sync, and before we handle the formation change for the new formation, which complicates the loop somewhat.

This is quite an edge case: two schedulers running jobs, one sees a new formation, starts a job, the other syncs with the cluster before seeing the new formation. Given there is a chance that this can happen, we shouldn't ignore it, but I wonder whether we should just add the job to the in-memory map and move on?